java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/entree/EntreeRepository.java

```diff
@@ -89,4 +89,16 @@ public List<Entree> findAllByIds(List<Integer> entreeIds) {
 
         return namedParameterJdbcTemplate.query(sql, parameters, new BeanPropertyRowMapper<>(Entree.class));
     }
+
+    public List<Entree> findEntreeThroughEntreeOrdered(Integer id) {
+
+        String sql = "SELECT entree.* " +
+                " FROM entree" +
+                " JOIN entree_ordered ON entree.id = entree_ordered.entree_id" +
+                " JOIN \"order\" ON \"order\".id = entree_ordered.order_id" +
+                " WHERE \"order\".id = ?";
```

Comment:
I believe this is ok, but I think you are unnecessarily joining the "order" table. You already know the "order".id, from the `entree_ordered` table. Can you just find your entrees from that instead?

Reply:
this has been condensed, thank you